proxy/proxy.go

```diff
@@ -119,6 +132,31 @@ func (proxy *Proxy) ListenAndServe() {
 	}
 }
 
+func unescapeBackSlashes(src string) string {
+	return escapedSlashRegexp.ReplaceAllString(src, "/")
+}
+
+func parseHostname(hostname string) (*regexp.Regexp, string, error) {
+	namedSubmatches := findStringNamedSubmatch(substitutionRegexp, hostname)
+	if len(namedSubmatches) < 1 {
+		errStr := fmt.Sprintf("Incorrect hostname expression: %s", hostname)
+		return nil, "", errors.New(errStr)
+	}
+
+	escapedPattern := namedSubmatches[substitutionPatternName]
+	escapedReplacement := namedSubmatches[substitutionReplacementName]
+	unescapedPattern := unescapeBackSlashes(escapedPattern)
+	unescapedReplacement := unescapeBackSlashes(escapedReplacement)
+
+	patternRegexp, err := regexp.Compile(unescapedPattern)
+	if err != nil {
+		errStr := fmt.Sprintf("Incorrect hostname pattern '%s': %s", escapedPattern, err.Error())
```

Comment:
My intention is to show `escapedPattern` to the user, which is the specific part of the substitution command which failed to compile. I could show the full hostname, but `Compile`'s error will refer exclusively to the pattern.

Thanks about `fmt.Errorf`, I will use it instead.